machine/v1beta1/types_awsprovider.go

```diff
@@ -91,6 +91,10 @@ type AWSMachineProviderConfig struct {
 	// +kubebuilder:validation:Maximum:=7
 	// +optional
 	PlacementGroupPartition *int32 `json:"placementGroupPartition,omitempty"`
+	// capacityReservationId specifies the target Capacity Reservation into which the instance should be launched.
+	// The field size should be greater than 0 and the field input must start with cr-***
+	// +optional
+	CapacityReservationID *string `json:"capacityReservationId,omitempty"`
```

Comment:
Does this need to be a pointer? Does it need omitempty? If it's the empty string we can deal with that as, customer has not specified an intent, but, since we marshal out the JSON, rendering the field would help with discoverability of the API

Reply:
Yes, we need to provide it as a pointer to the aws function input. 
Yes, it is not a mandatory field so I added omitempty.

Reply:
You can convert it to a pointer within the AWS actuator code

Optional fields in OpenShift configuration APIs don't necessarily have omitempty, we omit the omitempty so that the field gets rendered with the zero value. This helps users to discover the configuration options available to them